tests/test_build.py

```diff
@@ -321,3 +324,28 @@ def test_build_apps_collect_files_when_no_apps_built(tmp_path):
 
     assert os.path.exists('app_info.txt')
     assert os.path.exists('size_info.txt')
+
+
+@patch('idf_build_apps.app.subprocess_run')
```

Comment:
the error message is "Map file should be last argument". it's unclear what this test case is testing.

no need to use mock. you could write simple test case in `TestBuild` (idf.py is ensured) with real idf_size.py call.

one call without `extra_args`, see if json created with sections

one call with `extra_args`, see if json created with `--files`